lib/facebook/messenger/configuration/helpers.rb

```diff
@@ -0,0 +1,17 @@
+module Facebook
+  module Messenger
+    class Configuration
+      # The helpers module provides helpers that can be useful for
+      # configuration providers
+      module Helpers
+        def calculate_app_secret_proof(app_secret, access_token)
+          OpenSSL::HMAC.hexdigest(
```

Comment:
I would prefer to have this logic being extracted to separate standalone module, named e.g. AppSecretProofCalculator:

This Helpers module could be dropped and the logic placed at separate file:
```suggestion
module Facebook
  module Messenger
    class Configuration
      class AppSecretProofCalculator
        def self.call(app_secret, access_token)
          OpenSSL::HMAC.hexdigest(
            OpenSSL::Digest.new('SHA256'.freeze),
            app_secret,
            access_token
          )
        end
      end
    end
  end
end
```

And usage inside Facebook::Messenger::Configuration::Providers::Base is simple:

```suggestion
 Facebook::Messenger::Configuration::AppSecretProofCalculator.call(*memo_key)
```

Mainly suggesting it because noticed that actual logic of calculating the proof is not covered with tests. it is stubbed by expecting the any instance to receive instance method calculate_app_secret_proof.

My opinion is that it does not have to be defined as mixin. It could be a standalone class or module incapsulates the logic. Easy to use and easy to test.

Reply:
sure, sounds reasonable, I don't have much of an opinion either way.

I can put in a test for the new `AppSecretProofCalculator` but it seems a bit useless as I'd just be stubbing the single line method call within it. Do you think there's any point?

Reply:
addressed in latest commit.

Reply:
It depends on if you want to be specific saying i want to make sure we call exactly this logic to obtain value.
Otherwise someone could change the method to return some static string, e.g. '1', or replace it with SecureRandom.hex and many more. Tests still will pass.

Im saying that i would test at least we call OpenSSL::HMAC with digest calculated with 'SHA256'